src/utils/SuggestedCollabs.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @flow
+ * @file Suggested Collaborator utility functions
+ * @author Box
+ */
+
+import {
+    MAX_GAPS_FUZZY_MATCH,
+    MAX_SUGGESTIONS_TO_SHOW,
+    MIN_CHARACTERS_FOR_MATCHING,
+    SUGGESTED_COLLAB_CONTACT_TYPE,
+} from '../constants';
+import fuzzySearch from './fuzzySearch';
+
+/**
+ * Function to convert a suggested collaborator object to a contact object
+ *
+ * @param {SuggestedCollab} suggestedCollab The suggested collaborator object to convert
+ * @returns {Contact} The converted contact object
+ */
+function convertSuggestedCollabToContact(suggestedCollab: SuggestedCollab): Contact {
+    return {
+        email: suggestedCollab.email,
+        id: suggestedCollab.id,
+        name: suggestedCollab.name,
+        text: suggestedCollab.name,
+        type: SUGGESTED_COLLAB_CONTACT_TYPE,
+        value: suggestedCollab.email,
+    };
+}
+
+/**
+ * Function to compute suggested collaborators given a list of contacts and cached suggested collaborators.
+ * Used by input components to help augment API results with cached suggestions using a fuzzy search.
+ *
+ * @param {Array<Contact>} contacts Contacts returned from the server
+ * @param {SuggestedCollabLookup} suggestedCollaborators Cached suggested collab lookup to match on
+ * @param {string} inputValue Value the user typed into the input box searching for collabs
+ * @param {number} maxSuggestions Maximum number of items to suggest/return, default 3
+ * @param {number} minCharacters Minimum number of characters before searching, default 3
+ * @returns {Array<Contact>} Suggested collaborators formatted as Contacts in recommended order
+ */
+function computeSuggestedContacts(
+    contacts: Array<Contact>,
+    suggestedCollaborators: SuggestedCollabLookup,
+    inputValue: string,
+    maxSuggestions: number = MAX_SUGGESTIONS_TO_SHOW,
+    minCharacters: number = MIN_CHARACTERS_FOR_MATCHING,
+): Array<Contact> {
+    const exactMatchIds = contacts
+        .filter(option => {
+            const id = option.id.toString();
+            return id && suggestedCollaborators[id];
+        })
+        .map(option => option.id);
+
+    const suggestedCollabs = Object.keys(suggestedCollaborators).map(key => suggestedCollaborators[key]);
+
+    const suggestedExactMatches = suggestedCollabs.filter(option => {
```

Comment:
I may be reading this wrong, but it seems like this is very similar to the logic used to compute `exactMatchIds`. Can the two filters be combined and/or simplified, perhaps eliminating one of the arrays?